### PR TITLE
Update cache key used in rust CI script

### PR DESCRIPTION
**Pull request by findepi**
Cleanup of the GitHub workflow definition.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -62,8 +62,7 @@ jobs:
             ~/.cargo/git/db/
             ./target/
             ./datafusion-cli/target/
```
Comment (alamb) [inline .github/workflows/rust.yml:64]: Seems ok to me -- how do we know if this improves things or not 🤔

**Comment by alamb** [comment]
Thanks @findepi

**Comment by comphead** [comment]
can we have a description, what this PR is for?

**Comment by findepi** [comment]
@comphead thanks for reminder, added

**Review by comphead — APPROVED** [review]
it looks good for me thanks @findepi in terms of more concise text line
But I'm not sure what is the real benefit of it

**Comment by findepi** [comment]
> But I'm not sure what is the real benefit of it

technically none other than removing misleading comment & value

**Comment by alamb** [comment]
Thanks @findepi  and @comphead

**Pull request merged**